prime-router/src/main/kotlin/fhirengine/engine/FHIRConverter.kt

```diff
@@ -96,13 +122,16 @@ class FHIRConverter(
 
                 // 'stamp' observations with their condition code
                 bundle.getObservations().forEach {
+                    // TODO this is not quite right since there might be less bundles than items because of parsing and validating
```

Comment:
reminder to clean up comment and link ticket, but potentially wrong indices doesn't seem like something that should be put off?

Reply:
So I was leaning towards not fixing it since to get the index right would require moving where this happens which I think becomes a larger architectural change